applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1742,35 +1544,71 @@ int unicast_client_stop(uint8_t cig_index)
 	param.type = BT_CAP_SET_TYPE_AD_HOC;
 	param.release = true;
 
-	le_audio_event_publish(LE_AUDIO_EVT_NOT_STREAMING, NULL, 0);
+	le_audio_event_publish(LE_AUDIO_EVT_NOT_STREAMING, NULL, NULL, 0);
+
+	ret = srv_store_lock(CAP_PROCED_SEM_WAIT_TIME_MS);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to lock server store: %d", __func__, ret);
+		k_sem_give(&sem_cap_procedure_proceed);
+		return ret;
+	}
+
+	uint8_t num_servers = srv_store_num_get(true);
+
+	for (int i = 0; i < num_servers; i++) {
+		struct server_store *server = NULL;
 
-	for (int i = 0; i < ARRAY_SIZE(unicast_servers[cig_index][0]); i++) {
-		if (le_audio_ep_state_check(unicast_servers[cig_index][0][i].sink_ep,
-					    BT_BAP_EP_STATE_STREAMING)) {
-			/* Stop all sink streams currently in a streaming state */
-			streams[param.count++] = &unicast_servers[cig_index][0][i].cap_sink_stream;
+		ret = srv_store_server_get(&server, i);
+		if (ret) {
+			LOG_ERR("Failed to get server store for index %d: %d", i, ret);
+			continue;
 		}
 
-		if (le_audio_ep_state_check(unicast_servers[cig_index][0][i].source_ep,
-					    BT_BAP_EP_STATE_STREAMING)) {
-			/* Stop all source streams currently in a streaming state */
-			streams[param.count++] =
-				&unicast_servers[cig_index][0][i].cap_source_stream;
+		if (server->snk.num_eps > 0) {
+			for (int j = 0; j < server->snk.num_eps; j++) {
+				uint8_t state = BT_BAP_EP_STATE_IDLE;
+
+				ret = le_audio_ep_state_get(server->snk.eps[j], &state);
```

Comment:
If this returns an err, that is not clearly caught below.

Reply:
It is, on the line below.